pkg/trace/stats/sublayers.go

```diff
@@ -34,6 +43,181 @@ func (v SublayerValue) GoString() string {
 	return v.String()
 }
 
+// SublayerCalculator holds arrays used to compute sublayer metrics.
+// Re-using arrays reduces the number of allocations
+type SublayerCalculator struct {
+	activeSpans      []int
+	activeSpansIndex []int
+	openSpans        []bool
+	nChildren        []int
+	execDuration     []float64
+	parentIdx        []int
+	timestamps       timestampArray
+	maxSpans         int
+}
+
+// NewSublayerCalculator returns a new SublayerCalculator used to compute sublayer metrics
+func NewSublayerCalculator() *SublayerCalculator {
+	c := &SublayerCalculator{}
+	c.initFields(defaultSublayersCalculatorMaxSpans)
+	return c
+}
+
+// initFields initialized all arrays of the sublayer calculator
+// it should be called every time we receive a trace with more than maxSpans spans
+func (c *SublayerCalculator) initFields(maxSpans int) {
```

Comment:
You can remove this method and just call `reset` in the constructor. The name `init` could seem confusing when seen inside `reset`.

Reply:
I renamed `initFields` to `resize` and `reset` to `clear`. I think it makes more sense now.